spec/lib/tasks/populate_postcodes_spec.rb

```diff
@@ -0,0 +1,43 @@
+require "rake"
+require "spec_helper"
+
+Rails.application.load_tasks
+
+RSpec.describe Tasks do
+  describe "when running the task" do
+    before do
+      Postcode.destroy_all
+    end
+
+    let(:task) { Rake::Task["populate_postcodes_table"] }
+
+    it "should add postcodes in the database" do
+      expect(Postcode.count).to eq(0)
+
+      task.reenable
+      task.invoke("spec/lib/tasks/postcodes.csv")
+
+      expect(Postcode.count).to eq(3)
+    end
+
+    it "shouldn't fail when a postcode already exists" do
+      Postcode.create(postcode: "SE12TL")
+      expect(Postcode.count).to eq(1)
+
+      task.reenable
+      task.invoke("spec/lib/tasks/postcodes.csv")
+
+      expect(Postcode.count).to eq(3)
+    end
+
+    it "shouldn't overwrite existing postcode rows" do
+      Postcode.create(postcode: "SE12TL", results: '{ "foo": "bar" }')
+      expect(Postcode.count).to eq(1)
+
+      task.reenable
+      task.invoke("spec/lib/tasks/postcodes.csv")
+
+      expect(Postcode.where(postcode: "SE12TL").pluck(:results)).to eq(['{ "foo": "bar" }'])
```

Comment:
This can be tidied up to work with just one element instead of an array of results:

```suggestion
      expect(Postcode.find_by(postcode: "SE12TL").results).to eq(existing_result)
```